modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationCreationUtils.java

```diff
@@ -0,0 +1,253 @@
+   * @param connectionProviderResolver   the {@link ConnectionProviderResolver}
+   * @param configurationProviderFactory a {@link ConfigurationProviderFactory} used to instantiate the provider
+   * @param expressionManager            an {@link ExpressionManager}
+   * @param reflectionCache              a {@link ReflectionCache}
+   * @param parametersOwner              the name of the owning component or an empty string
+   * @param dslSyntaxResolver            a {@link DslSyntaxResolver}
+   * @param extensionClassLoader         the extension's {@link ClassLoader}
+   * @param muleContext                  the current {@link MuleContext}
+   * @return a new {@link ConfigurationProvider}
+   * @throws MuleException if the resolver cannot be created
+   * @since 4.5.0
+   */
+  public static ConfigurationProvider createConfigurationProvider(ExtensionModel extensionModel,
+                                                                  ConfigurationModel configurationModel,
+                                                                  String configName,
+                                                                  Map<String, Object> parameters,
+                                                                  Optional<ExpirationPolicy> expirationPolicy,
+                                                                  ConnectionProviderValueResolver connectionProviderResolver,
+                                                                  ConfigurationProviderFactory configurationProviderFactory,
+                                                                  ExpressionManager expressionManager,
+                                                                  ReflectionCache reflectionCache,
+                                                                  String parametersOwner,
+                                                                  DslSyntaxResolver dslSyntaxResolver,
+                                                                  ClassLoader extensionClassLoader,
+                                                                  MuleContext muleContext) {
+    return withContextClassLoader(extensionClassLoader, () -> {
+      ResolverSet resolverSet = getResolverSetFromStaticValues(
+                                                               configurationModel,
+                                                               parameters,
+                                                               muleContext,
+                                                               false,
+                                                               reflectionCache,
+                                                               expressionManager,
+                                                               parametersOwner,
+                                                               dslSyntaxResolver);
```

Comment:
do we have a TODO to remove this code dependency on the dsl?

Reply:
That's way too embedded into the logic and needed to support some edge cases. We're not looking at changing that at the moment

Reply:
do we have a pending task to not lose track of it and do it sometime?